internal/modelconfig/types/configuration_test.go

```diff
@@ -0,0 +1,30 @@
+package types
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/go-enry/go-enry/v2/regex"
+)
+
+func TestClientSideProviderConfigMarshalJSON(t *testing.T) {
+	config := ClientSideProviderConfig{}
+
+	jsonBytes, err := json.Marshal(config)
+	if err != nil {
+		t.Fatalf("Failed to marshal ServerSideProviderConfig: %v", err)
+	}
+
+	actual := string(jsonBytes)
+
+	// Check whether ClientSideProviderConfig contains any data which looks sensitive
```

Comment:
Great idea, and A++ for adding the test! 👏👏👏

I, being a broken human being, cringe a little bit seeing a comment that is a complete sentence but not ending in a period. (😬)

```suggestion
	// Check whether ClientSideProviderConfig contains any data which looks sensitive.
	// This isn't perfect, but provides some sanity-check level coverage that we haven't done anything obviously wrong.
```